### PR TITLE
feat(vite): allow passing multiple test files to vite:test

### PR DESCRIPTION
**Pull request by jwalton9**
## Current Behavior

You can only pass one testfile to run to `vite:test` 

## Expected Behavior

Able to pass multiple test files to run to `vite:test` this is useful for CI for example when using test splitting

### CODE REVIEW

File: packages/vite/src/migrations/update-16-4-0-update-test-file-config/update-16-4-0-test-file-config.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+          if (target.configurations) {
+            for (const configuration of Object.values(target.configurations)) {
+              if (configuration.testFile) {
+                configuration.testFile = [configuration.testFile];
+              }
+            }
+          }
+        }
+      }
```
Comment (barbados-clemens) [inline packages/vite/src/migrations/update-16-4-0-update-test-file-config/update-16-4-0-test-file-config.ts:23]: You can use `forEachExecutorOption` instead.

```ts
// might just be from '@nx/devkit' now?
import { forEachExecutorOptions } from '@nx/devkit/src/generators/executor-options-utils';

const projects = getProjects(tree);

forEachExecutorOptions<VitestExecutorOptions>(tree, '@nx/vite:test', (options, projectName, targetName, configuration) => {
  const projectConfig = projects.get(projectName);

  // update testFile option if preset

  updateProjectConfiguration(tree, projectName, projectConfig)
})
```

File: packages/vite/src/migrations/update-16-4-0-update-test-file-config/update-16-4-0-test-file-config.spec.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  beforeEach(() => {
+    tree = createTreeWithEmptyWorkspace({ layout: 'apps-libs' });
+    addProjectConfiguration(tree, 'vitest', {
+      root: 'vitest',
+      targets: {
+        test: {
+          executor: '@nx/vite:test',
+          options: {
+            testFile: 'test-file.ts',
+          },
```
Comment (barbados-clemens) [inline packages/vite/src/migrations/update-16-4-0-update-test-file-config/update-16-4-0-test-file-config.spec.ts:22]: you should add a test that makes sure it doesn't modify a project that doesn't have a `testFile` option.

File: packages/vite/src/executors/test/schema.d.ts [hunk 3]
```diff
@@ -8,5 +8,5 @@ export interface VitestExecutorOptions {
   update?: boolean;
   reportsDirectory?: string;
   coverage?: boolean;
-  testFile?: string;
+  testFile?: string[];
```
Comment (barbados-clemens) [inline packages/vite/src/executors/test/schema.d.ts:11]: now that this is an array does it make sense to rename this to be plural? `testFiles` 🤔

File: packages/vite/migrations.json [hunk 4]
```diff
@@ -23,6 +23,12 @@
       "version": "16.0.0-beta.1",
       "description": "Replace @nrwl/vite with @nx/vite",
       "implementation": "./src/migrations/update-16-0-0-add-nx-packages/update-16-0-0-add-nx-packages"
+    },
+    "update-16-4-0-test-file-config": {
+      "version": "16.4.0-beta.2",
```
Comment (barbados-clemens) [inline packages/vite/migrations.json:28]: you'll want to update this to [16.4.0-beta.5](https://www.npmjs.com/package/nx/v/16.4.0-beta.5)

File: packages/vite/src/migrations/update-16-4-0-update-test-file-config/update-16-4-0-test-file-config.ts [hunk 5]
```diff
@@ -0,0 +1,28 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
```
Comment (barbados-clemens) [inline packages/vite/src/migrations/update-16-4-0-update-test-file-config/update-16-4-0-test-file-config.ts:1]: remove eslint comment

**Review by barbados-clemens — COMMENTED** [review]
Hi thanks for the PR, just a couple things that could be improved!

File: packages/vite/src/executors/test/schema.d.ts (hunk 3 above)
Comment (jwalton9) [reply packages/vite/src/executors/test/schema.d.ts:11]: I'm not sure. I think this makes sense when using the option in `project.json` e.g.
```json
{
   "testFiles": ["tests/foo.test.tsx", "tests/bar.test.tsx"]
}
```
But I prefer `testFile` when using the CLI
```
nx run app:test --testFIle=foo.test.tsx --testFile=bar.test.tsx
```
over 
```
nx run app:test --testFIles=foo.test.tsx --testFiles=bar.test.tsx
```

It just personal preference, so happy to go with `testFiles` but just making the case for keeping as is

Comment (jwalton9) [reply packages/vite/src/executors/test/schema.d.ts]: Great, thanks for the suggestion, I've updated to use `testFiles` with a `testFile` alias

**Review by barbados-clemens — APPROVED** [review]

**Comment by barbados-clemens** [comment]
updated the tests to handle the option rename and updated  the target for the migration

File: packages/vite/migrations.json [hunk 6]
```diff
@@ -23,6 +23,12 @@
       "version": "16.0.0-beta.1",
       "description": "Replace @nrwl/vite with @nx/vite",
       "implementation": "./src/migrations/update-16-0-0-add-nx-packages/update-16-0-0-add-nx-packages"
+    },
+    "update-16-4-1-test-file-config": {
+      "version": "16.4.1-beta.0",
```
Comment (ndcunningham) [inline packages/vite/migrations.json:28]: Nx 16.4 is out so this would an update

**Review by ndcunningham — CHANGES_REQUESTED** [review]

**Review by ndcunningham — APPROVED** [review]

**Pull request merged**